### DISCUSSION TITLE
transmission-daemon 3.0 build issues

### DISCUSSION THREAD

**Discussion by xMinerv** [transmission/transmission#4370] [Support] [answered]
Hi there. I have compiled transmission v3.00 on both Rocky Linux 9.1 and Ubuntu 22.04.1 LTS, with the same results. When running transmission-daemon under systemd, the website will be up and active but when i upload a .torrent file, transmission-daemon will close its self:
Started Transmission BitTorrent Daemon. 
segfault at 0 ip 00007f1a32ee7c25 sp 00007f1a3186a0c0 error 4 in libcrypto.so.3[7f1a32e07000+25d000] 
Code: 00 00 00 00 90 f3 0f 1e fa 41 54 41 89 f4 55 48 89 fd 48 81 ec a8 00 00 00 64 48 8b 04 25 28 00 00 00 48 89 84 24 98 00 00 00 <48> 8b 07 48 83 78 78 00 0f 84 cd 00 00 00 66 0f ef c0 48 63 c6 48 
transmission.service: Main process exited, code=killed, status=11/SEGV 
transmission.service: Failed with result 'signal'. 

It was compiled using these flags:
cmake -DCMAKE_INSTALL_PREFIX=/opt/transmission-3.00 -DCMAKE_BUILD_TYPE=RelWithDebInfo
for Ubuntu, I have installed using libcurl4-openssl-dev:amd64 7.81.0-1ubuntu1.6 and libssl-dev:amd64 3.0.2-0ubuntu1.7
for RockyLinux, I have installed libcurl-devel-7.76.1-19.el9.x86_64 and openssl-devel-3.0.1-43.el9_0.x86_64

How should i go about building transmission 3 for success?

Thank you!

**Comment by ckerr** [comment, marked as answer]
Without a crash report or stacktrace there's no way to know where it's crashed. So if you're wanting to triage 3.00 yourself, you'll want to run transmission in a debugger and get a trace after it crashes.

As a practical matter though, 3.00 is several years old and is several hundred commits behind 4.0.0 which is in its beta cycle. At this point  bugs reported against 4.0.0 are *much* more likely to be addressed so I'd suggest building & testing with a 4 beta instead.

**Comment by gramakri** [comment]
Same issue with using transmission 3.0 on ubuntu 22.04:

dmesg has this, I will try to get a proper back trace. I have to fixup my docker environment to allow debugging.

```
[3361823.908931] transmission-da[4034037]: segfault at 0 ip 00007fd250ad1c25 sp 00007fd24f4ee0c0 error 4 in libcrypto.so.3[7fd2509f1000+25d000]
[3361823.908950] Code: 00 00 00 00 90 f3 0f 1e fa 41 54 41 89 f4 55 48 89 fd 48 81 ec a8 00 00 00 64 48 8b 04 25 28 00 00 00 48 89 84 24 98 00 00 00 <48> 8b 07 48 83 78 78 00 0f 84 cd 00 00 00 66 0f ef c0 48 63 c6 48
```

**Comment by qianbinbin** [comment]
This happened when I was requesting the daemon v3.00 to update the blocklist: 
```
transmission-remote:  (http://127.0.0.1:9091/transmission/rpc/) Server returned nothing (no headers, no data)
```

The log of the daemon was (unfortunately the message-level was set to 1):

```
systemd[1]: transmission-daemon.service: Main process exited, code=killed, status=11/SEGV
systemd[1]: transmission-daemon.service: Failed with result 'signal'.
```

Related dmesg:

```
[39789.055320] show_signal_msg: 1 callbacks suppressed
[39789.055323] transmission-da[3401]: segfault at 31 ip 00007f46b98df5d3 sp 00007f46b7bfe4e0 error 6 in libevent-2.1.so.7.0.1[7f46b98b5000+31000] likely on CPU 2 (core 0, socket 0)
[39789.055332] Code: ff 48 83 c4 08 c3 90 55 53 48 83 ec 08 48 8b 1f 48 85 db 74 52 48 89 fd eb 36 0f 1f 44 00 00 48 89 50 08 48 8b 43 08 48 8b 13 <48> 89 10 48 8b 7b 10 e8 21 69 fd ff 48 8b 7b 18 e8 18 69 fd ff 48
```

And core dump was not enabled.
